File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int min = nums[0];

        int l = 0;
        int r = nums.length - 1;

        while(l<=r){
            if(nums[l] < nums[r]){
                return Math.min(min, nums[l]);
            }

            int m = (l+r)/2;
            System.out.println(m);
            if(nums[l] <= nums[m]){
                min = Math.min(min, nums[m]);
                l = m+1;
            }else{
                min = Math.min(min, nums[m]);
                r = m-1;
            }
        }

        return min;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zerocount = 0;
        zeroindex = 0;
        res = 1;

        for i, n in enumerate(nums):
            if n != 0:
                res *= n
            else:
                zerocount += 1
                zeroindex = i
            
            if zerocount > 1:
                return [0] * len(nums)
        
        resList = []
        if zerocount == 1: 
            resList = [0] * len(nums)
            resList[zeroindex] = res
            return resList
            


        for n in nums:
            if n != 0:
                resList.append(int(res / n))
            else:
                resList.append(res)

        return resList


        